woocommerce/google-listings-and-ads
# Settings page - disconnect accounts modal

### Changes proposed in this Pull Request:

Closes #286 .

Implement the rest parts of #286 
- Implement the model components for proceeding with the accounts disconnection
- Call disconnection APIs and then re-fetch data or relocate page according to which disconnecting action
- Tracking events when accounted are disconnected

Not in the scope of the #286, but related to it
- Make the size of the **\<WarningIcon\>** component configurable
- Fix that `undefined` may appear on the CSS class of the **\<AppModel\>** component

### Screenshots:

#### 🎥 . Disconnect Google ads account then disconnect all accounts

![Kapture 2021-03-29 at 19 43 57](https://user-images.githubusercontent.com/17420811/112833720-b4bc5600-90c9-11eb-88bb-a4cd24595973.gif)

#### 🎥 . Disconnect all accounts including Google ads account

![Kapture 2021-03-29 at 19 50 24](https://user-images.githubusercontent.com/17420811/112833735-b9810a00-90c9-11eb-9fe8-425c251ba600.gif)

#### 📷 . The modal would disable controllable elements and unable to dismiss during disconnection

![image](https://user-images.githubusercontent.com/17420811/112925887-30a7b400-9145-11eb-880e-78cd326c1c59.png)

### Detailed test instructions:

1. Open the DevTool console and execute `localStorage.setItem( 'debug', 'wc-admin:*' );` to enable tracking debugging mode, and tick the **Preserve log** option from settings
2. Head to the GLA Settings page: `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings`
3. Disconnect the Google ads account
    1. The Google ads account should be disappeared from the section of the Setting page
4. Disconnect all accounts
    1. The page should be redirected to the Get Started page
5. When communicating with APIs, the modal should be unable to dismiss. This part may not be easy to verify due to the APIs response quickly. To simulate it through disconnecting ads account, you can add a code [at the beginning of here](https://github.com/woocommerce/google-listings-and-ads/blob/52dacc66b075ae2ba2354cc49b705973a3b3749d/js/src/data/actions.js#L373) to skip API call and lock 10 sec:
    ```js
    export function* fetchGoogleAdsAccountBillingStatus() {
    	// Add this line
    	return new Promise( ( r ) => setTimeout( r, 10000 ) );

    	try {
    		const response = yield apiFetch( {
    ```
6. Relevant event should be logged on the DevTool console
